sensor/common/compliance/nodescan_handler_impl.go

```diff
@@ -6,35 +6,51 @@ import (
 	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/centralsensor"
 	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
 type nodeScanHandlerImpl struct {
 	nodeScans <-chan *storage.NodeScanV2
-	toCentral chan *central.MsgFromSensor
+	toCentral <-chan *central.MsgFromSensor
+	// toCentralMux prevents the race condition between Start() [writer] and ResponsesC() [reader]
+	toCentralMux *sync.Mutex
 
-	stopC    concurrency.ErrorSignal
-	stoppedC concurrency.ErrorSignal
+	stopC concurrency.ErrorSignal
+	// runFinishedC is signaled when the goroutine inside of run() finishes
+	runFinishedC concurrency.ErrorSignal
```

Comment:
`stoppedC` is more established name for this thing than `runFinishedC` in Sensor's codebase. I suggest to stick to that.

That said, I recall that you agreed to delete `Stopped()` in the last PR, but restored it here. I wonder why do you need it again?
https://github.com/stackrox/stackrox/pull/3533#discussion_r1015421998

Reply:
I decided to add it again, so that the tests can wait for the component to fully stop. Waiting is important because we run a goroutine-leak tests in a deferred statement and it happened sometimes that the goroutines were analyzed before the component stopped.

I will restore the name `stoppedC`.